### ISSUE TITLE
Shared app launcher icon for both platforms

### ISSUE THREAD

**Issue by MarkOSullivan94** [flutter/flutter#13293]
Most apps share the icon they use for both platforms and so it would be great if it was possible to specify an app launcher icon for both the iOS project and Android project without having to delve into both projects and adding image files in two separate locations. 

I think the current way is specified within this Stack Overflow question: https://stackoverflow.com/questions/43928702/how-to-change-the-application-icon-on-flutter

The assets and images guide describes how you can have shared assets among both projects, perhaps this would be the best place to include shared app launcher icons? Default projects could come with this asset directory preloaded with the default flutter icon which is used as an app launcher icon for both apps.

Perhaps for the very few cases where developers want to specify slightly different app launcher icons, you could configure the `pubspec.yaml` so that you can declare if you want a shared app launcher icon or if not, could specify what app icon from the asset directory that you want for each platform.

```
# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  # Line for having a shared app launcher icon
  uses-shared-app-icon: true

  # if above line isn't used then the next two lines would be required to supply icon for both platforms
  android-app-icon: 'assets/android-icon/*'
  ios-app-icon: 'assets/ios-icon/*

```

Not sure if this idea would be a good way to specify the shared icons but having the ability to do so would be greatly appreciated.

**Comment by mravn-google** [comment]
cc @szakarias

**Comment by eseidelGoogle** [comment]
I'm not sure you want this.  The UI guidelines for the two platforms are different.

There are however a variety of tools which can help you generate the two sets of icons from a single  source image.

e.g. https://romannurik.github.io/AndroidAssetStudio/ or https://itunes.apple.com/us/app/iconkit-the-icon-resizer/id507135296?mt=12 are ones I've used in the past.

**Comment by MarkOSullivan94** [comment]
@eseidelGoogle what do you mean? 

The UI guidelines for the app design are different yes but in terms of the icon which the user clicks on, it's the same for both platforms unless you're specifying different android icons for the adaptive icons in Oreo or above. 

In that case you could create a directory within the assets/android-icon/ directory and have one for each or the adaptive icon shapes.

Otherwise I don't see why you couldn't just use the identical icon for both.

**Comment by eseidelGoogle** [comment]
As an example, here are the launcher icons for stocks for iOS:
https://github.com/flutter/flutter/blob/master/examples/stocks/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-60%402x.png
https://developer.apple.com/ios/human-interface-guidelines/icons-and-images/app-icon/
And Android:
https://github.com/flutter/flutter/blob/master/examples/stocks/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png
https://developer.android.com/guide/practices/ui_guidelines/icon_design.html

The size, the borders (or lack there of) as well as shading are all expected to be different.  And that's not even considering round icons for Android, or the various sizes which need to be generated for both.

These sorts of generations are definitely automatable.  Just not sure Flutter itself has a lot of value add here (at least not in the short term).

**Comment by jacoplane** [comment]
Ionic has feature:

http://blog.ionicframework.com/automating-icons-and-splash-screens/

**Comment by sethladd** [comment]
Is this something someone could build a simple/small script to handle?

The pubspec.yaml file is designed to be extensible, so you (and others) can add more custom metadata there.

Then, you could write the Dart code to read the custom metadata from pubspec.yaml and "do the right thing" with the image assets. Then, you could publish that util script to pub.

Users can install your package, and then run it from the command line (because pub, when it installs packages, can put package bin scripts into your path).

Then, your users can run your script via some build tool or task runner.

Basically, I think you don't necessarily _require_ the flutter tool to perform these kinds of asset manipulations and managements. I think what's required is for us to help you contribute this kind of functionality to the ecosystem. And we're very happy to do that.

Thoughts?

**Comment by franzsilva** [comment]
Im guessing you could use something like this. 
https://pub.dartlang.org/packages/image
Then use that as a base to generate the images.

**Comment by franzsilva** [comment]
@MarkOSullivan94 , would you like to create a package like this to contribute?, I would be more than happy to help.

**Comment by franzsilva** [comment]
Starting work at https://github.com/franzsilva/flutter_launcher_icons

**Comment by MarkOSullivan94** [comment]
Added changes which should allow for generation of iOS icons and allow you to specify what platform you want to generate the icons for. Just waiting for @franzsilva to test it out and then merge the pull requests into the repo.

**Comment by franzsilva** [comment]
Done Merging the changes. We have a 0.1.0 version of the plugin. I think this issue can be closed. it basically does what was originally intended. Future version POC to generate formated images like round, square with shadow, padding is in the works.

**Comment by franzsilva** [comment]
https://pub.dartlang.org/packages/flutter_launcher_icons
